### ISSUE TITLE
Update `REGEXP_MATCH` scalar function to support Utf8View

### ISSUE THREAD

**Issue by tshauck** [apache/datafusion#11911]

**Comment by PsiACE** [comment]
take

**Comment by Omega359** [comment]
Note that I just saw that https://github.com/apache/arrow-rs/issues/6370 was merged into arrow-rs which may be relevant to this work

**Comment by alamb** [comment]
I think we now have the needed upstream support in arrow-rs for this but it will not be available until arrow 54.1.0 -- slated to be released this week

See commit https://github.com/apache/arrow-rs/commit/5249f99e0e6671fc06996365dd4cf22a8427348c

**Comment by alamb** [comment]
@Weijun-H  updated us `54.1.0` today in 
- https://github.com/apache/datafusion/pull/14415

We are ready to rock on this PR

**Comment by Omega359** [comment]
I am working on this now. Code changes are pretty simple, the .slt test needs to be refactored to reflect the changes made to the strings slt tests for improved readability

**Comment by alamb** [comment]
Thanks @Omega359  -- I will review it with urgency!

You probably already found it but if you put the tests in here: 
https://github.com/apache/datafusion/tree/main/datafusion/sqllogictest/test_files/string

You'll end up with coverage for all the different string types with a single set of tests

**Comment by Omega359** [comment]
Yup. I'm actually just going to put them in regexp.slt for now and file a PR to refactor it instead just to get this in as quick as possible.
